centos-7-minimal.ks

```diff
@@ -65,13 +75,52 @@ shim
 LANG="en_US"
 echo "%_install_lang $LANG" > /etc/rpm/macros.image-language-conf
 
+# sudo permission
+echo "%docker ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers.d/docker
+sed -i "s/^.*requiretty/#Defaults requiretty/" /etc/sudoers
+
+cat > /etc/rc.d/init.d/copy-ssh << EOF
+#!/bin/sh
+
+# TODO Generate Locale. This needs further investigation. I'd prefer to disable
+# 'AcceptEnv LANG LC_*'' in '/etc/ssh/sshd_config', but it did not work initially
+localedef -v -c -i en_US -f UTF-8 en_US.UTF-8
+
+# TODO Why can I not generate the docker user outside of this init script as part
+# of the kickstart file?
+useradd -p tcuser docker
+groupadd docker
+usermod -a -G docker docker
+# TODO Need to make sure to have /sbin on the PATH. Is there a better way?
+# http://stackoverflow.com/questions/19983710/some-commands-not-wroking-on-remote-servers-through-ssh-shell
+echo 'PATH=$PATH:/sbin' >> /home/docker/.bashrc
+
+# Read /userdata.tar with ssh keys
+# TODO we need to add checks whether the userdata.tar is still there or whether
+# the disk got already formatted
+dd if=/dev/sda of=/userdata.tar bs=1 count=4096 2>/dev/null
+tar xf /userdata.tar -C /home/docker/ > /var/log/userdata.log 2>&1
+rm -f '/home/docker/boot2docker, please format-me'
+chown -R docker:docker /home/docker/.ssh
+EOF
+
+chmod +x /etc/rc.d/init.d/copy-ssh
+/sbin/restorecon /etc/rc.d/init.d/copy-ssh
```

Comment:
Is the restorecon is done to make it future proof? Because we disable selinux beginning of the kickstart file.